PublishedModules/Psychokiller1888/FreeCurrencyConverterDotCom/FreeCurrencyConverterDotCom.py

```diff
@@ -22,50 +21,41 @@ def __init__(self):
 			(self._INTENT_ANSWER_CURRENCY, self.convertCurrencyIntent),
 			(self._INTENT_CONVERT_CURRENCY, self.convertCurrencyIntent),
 		]
-
+		
+		self._apiKey = self.getConfig('apiKey')
 		super().__init__(self._SUPPORTED_INTENTS)
 
 
+	@online
 	def convertCurrencyIntent(self, session: DialogSession, **_kwargs):
-		amount = 1
-		if 'Amount' in session.slots:
-			amount = session.slotsAsObjects['Amount'][0].value['value']
-		elif 'amount' in session.customData:
-			amount = session.customData['Amount']
-
-		toCurrency = self.ConfigManager.getAliceConfigByName('baseCurrency', self.name)
-		if 'ToCurrency' in session.slotsAsObjects:
-			toCurrency = session.slotsAsObjects['ToCurrency'][0].value['value']
-		elif 'toCurrency' in session.customData:
-			toCurrency = session.customData['toCurrency']
-
-		if 'FromCurrency' not in session.slots:
-			if 'Currency' in session.slots:
-				fromCurrency = session.slotsAsObjects['Currency'][0].value['value']
-			else:
-				self.continueDialog(
-					sessionId=session.sessionId,
-					intentFilter=[self._INTENT_ANSWER_CURRENCY],
-					text=self.TalkManager.randomTalk(module=self.name, talk='fromWhatCurrency'),
-					customData={
-						'module'    : self.name,
-						'amount'    : amount,
-						'toCurrency': toCurrency
-					}
-				)
-				return
-		else:
-			fromCurrency = session.slotsAsObjects['FromCurrency'][0].value['value']
+		amount = session.slots.get('Amount', session.customData.get('Amount', 1))
+
+		toCurrency = session.slots.get('ToCurrency',
+			session.customData.get('toCurrency',
+			self.ConfigManager.getAliceConfigByName('baseCurrency', self.name)))
+
+		fromCurrency = session.slots.get('FromCurrency', session.slots.get('Currency'))
+		if not fromCurrency:
+			self.continueDialog(
+				sessionId=session.sessionId,
+				intentFilter=[self._INTENT_ANSWER_CURRENCY],
+				text=self.TalkManager.randomTalk(module=self.name, talk='fromWhatCurrency'),
+				customData={
+					'module'    : self.name,
+					'amount'    : amount,
+					'toCurrency': toCurrency
+				}
+			)
+			return
 
 		try:
-			url = f"https://free.currconv.com/api/v7/convert?q={fromCurrency}_{toCurrency}&compact=ultra&apiKey={self.getConfig('apiKey')}"
-			req = requests.get(url=url)
-			data = json.loads(req.content.decode())
+			url = f'https://free.currconv.com/api/v7/convert?q={fromCurrency}_{toCurrency}&compact=ultra&apiKey={self._apiKey}'
+			data = requests.get(url=url).json()
 
 			conversion = data[f'{fromCurrency}_{toCurrency}']
 			converted = round(float(amount) * float(conversion), 2)
 
-			self.endDialog(session.sessionId, text=self.randomTalk('answer').format(amount, fromCurrency, converted, toCurrency), siteId=session.siteId)
+			self.endDialog(session.sessionId, text=self.randomTalk('answer').format(amount, fromCurrency, converted, toCurrency))
 		except Exception as e:
```

Comment:
since this catches all exceptions, no exceptions will get caught by the online decorator